Animations/CMakeLists.txt

```diff
@@ -0,0 +1,220 @@
+# Configure vendored libraries
+set(DEAR_IMGUI_USER_CONFIG "${CMAKE_CURRENT_LIST_DIR}/include/core/InternalImGuiConfig.h")
+set(DEAR_IMGUI_EXTRA_INCLUDE_DIRS
+        "${CMAKE_CURRENT_LIST_DIR}/vendor/cppUtils/single_include"
+        "${CMAKE_CURRENT_LIST_DIR}/include")
+
+# Add all vendored libraries
+add_subdirectory(vendor)
+
+##############
+# Animations #
+##############
+# TODO: Project should be restructured to avoid accessing files outside the scope
+set(MATH_ANIMATIONS_ASSETS_DIR "${CMAKE_CURRENT_LIST_DIR}/../assets")
+set(MATH_ANIMATIONS_GRAMMARS_DIR "${CMAKE_CURRENT_LIST_DIR}/vendor/grammars")
+set(MATH_ANIMATIONS_THEMES_DIR "${CMAKE_CURRENT_LIST_DIR}/vendor/themes")
+
+set(MATH_ANIMATIONS_INCLUDE_DIR "${CMAKE_CURRENT_LIST_DIR}/include")
+
+set(MATH_ANIMATIONS_SOURCE
+        src/multithreading/GlobalThreadPool.cpp
```

Comment:
A couple of notes here. 

I know that CMake recommends not GLOBing. However, it's 2022, and CMake's reasons for disallowing this make no sense to me whatsoever. From what I understand, they don't want you to GLOB the files because then CMake won't be able to tell which files were added on subsequent `cmake ..` calls, which means it will require a full rebuild. First, like I said, it's 2022 and the fact that CMake refuses to add a cache to handle this is baffling to me. There's no reason a wildcard shouldn't be accepted because they don't want to add a simple cache telling them which files were used in the previous build, but maybe there's deeper reasons then that.

Secocnd, I don't care about this because I only ever add files through the IDE, which doesn't require me to re-run `cmake ..` anyways, and I will inevitably forget to add the source file here because I'm adding it through the IDE.

Last, I notice that the header files aren't listed here. In Visual Studio, if the header files aren't included in the source then I can't grep through the header files using the IDE which is a massive pain point. So if the files are going to be listed out manually then I would need the headers here as well.

If there's some other huge reason that I should be avoiding GLOB, then please let me know. But if the reasons I outlined above are the main concerns, then I would really prefer to GLOB since I'll be the primary developer on this project and this suits my development style the most :)

Reply:
Ohhh, thats a hard one. The reason I do it this way is because your way works only for Visual Studio. Adding source files without re-running CMake only works for Visual Studio (and maybe XCode?) because these IDE's have their own project systems.

However, if you are using Vim or CLion, you need to re-run CMake in order to add new files. These IDE's (or editors) rely in CMake to provide the source file context.

Beyond that... yes, it's super stupid, but globbing usually does break. I've kind of gotten numb and accepted it as a matter of fact that you have to do that with CMake.

For this project I thus think we should use the official recommendations and add source files manually (additionally that allows adding platform specific sources). I know it sucks, but for a lot of IDE's you do have to re-run `cmake ..` when you change stuff.

Reply:
Hey Janrupf, if this is the biggest reason for the recommendations, then I'm gonna ask that we change this back to globbing then. As I'll be the one maintaining this for the long run, it will be easier if the build files suit my development workflow more. If there begins to be a large enough group of contributors that don't use Visual Studio and they are actively adding lots of files and having problems because of glob, then we can revisit this. But for now I'd like to leave the glob :)